tests/test_actions.py

```diff
@@ -0,0 +1,37 @@
+"""Tests the base functionality of the library."""
+
+
+from action_msgs.msg import GoalStatus
+from example_interfaces.action import Fibonacci
+from rclpy.action.client import ClientGoalHandle
+
+# What we are testing
+from ros2_easy_test import ROS2TestEnvironment, with_single_node
+
+# Module under test and interfaces
+from .example_nodes.minimal_action_server_with_context import MinimalActionServerWithContext
+
+
+@with_single_node(MinimalActionServerWithContext)
+def test_fibonacci_action(env: ROS2TestEnvironment) -> None:
```

Comment:
Since we share the `ActionObjects` instance across calls to the same action, we should test calling the same action twice. Do we really want to change that object, anyways, or just the client?

Reply:
So if I run this test twice it currently fails. Even if I repeat it as a separate test_fibonnaci_action2(), it remembers the previous history. I copied the imlementation from get_service.  Doesn't that face the same issue? How do you want to address this?  
1. Should we creating a unique string for the dict which holds _ActionObjects.  
2. Alternatively we can always start with an empty _ActionObjects?  I could replace :
```
        with self._registered_actions_lock:
            try:
                return self._registered_actions[name]
            except KeyError:
                objects = _ActionObjects(
                    client=ActionClient(self, action_type, name, callback_group=self._action_cb_group)
                )
                self._registered_actions[name] = objects
                return objects
```
with:
```
        with self._registered_actions_lock:
            objects = _ActionObjects(
                client=ActionClient(self, action_type, name, callback_group=self._action_cb_group)
            )
            self._registered_actions[name] = objects
            return objects

```
WDYT?

Reply:
I think the problem is not to share the `ActionClient`, but to share the custom `ActionObjects` instance with the other fields in it. So can't we just store the client in `self._registered_actions` (maybe renamed), and create a new `ActionObjects` locally in `def send_action_goal_and_wait_for_response(...)`? I'm not too familiar with actions but I think that would work?

Reply:
I do not understand what you are suggesting exactly.  I think the option 2 I suggested above is simple/cleaner.  Have implemented that for now.

Reply:
Okay, I'll try to phrase it differently:

`ActionObjects` have state. They correspont to a specific call to run the action.

`ActionClient` on the other hand, can be called multiple times, and do not correspont to only a single call of the action.

We should, therefore, not store the `ActionObjects` at all in the environment. It shall only exist locally in the function handling the call. In `self._registered_actions`, we should instead only store what can be shared across calls, which is the `ActionClient`.

I am talking about correctness, not simplicity. :)

Reply:
Done.